exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/Serializer.java

```diff
@@ -0,0 +1,416 @@
+    for (int i = 0; i < s.length(); i++) {
+      char c = s.charAt(i);
+      switch (c) {
+        case '\\':
+          writer.write("\\\\");
+          break;
+        case '\"':
+          writer.write("\\\"");
+          break;
+        case '\n':
+          writer.write("\\n");
+          break;
+        default:
+          writer.write(c);
+      }
+    }
+  }
+
+  // TODO(anuraaga): Can likely be optimized especially for the already-sanitized case.
+  private static String sanitizeMetricName(String metricName) {
+    return SANITIZE_BODY_PATTERN
+        .matcher(SANITIZE_PREFIX_PATTERN.matcher(metricName).replaceFirst("_"))
+        .replaceAll("_");
+  }
+
+  static class Prometheus004Serializer extends Serializer {
+
+    static final Serializer INSTANCE = new Prometheus004Serializer();
+
+    @Override
+    String headerName(String name, PrometheusType type) {
+      if (type == PrometheusType.COUNTER) {
+        return name + "_total";
+      }
+      return name;
+    }
+
+    @Override
+    void writeHelp(Writer writer, String help) throws IOException {
+      for (int i = 0; i < help.length(); i++) {
+        char c = help.charAt(i);
+        switch (c) {
+          case '\\':
+            writer.write("\\\\");
+            break;
+          case '\n':
+            writer.write("\\n");
+            break;
+          default:
+            writer.write(c);
+        }
+      }
+    }
+
+    @Override
+    void writeTimestamp(Writer writer, long timestampNanos) throws IOException {
+      writer.write(Long.toString(TimeUnit.NANOSECONDS.toMillis(timestampNanos)));
+    }
+
+    @Override
+    void writeExemplar(
+        Writer writer, Collection<ExemplarData> exemplars, double minExemplar, double maxExemplar) {
+      // Don't write exemplars
+    }
+
+    @Override
+    void writeEof(Writer writer) {
+      // Don't write EOF
+    }
+  }
+
+  static class OpenMetrics100Serializer extends Serializer {
+
+    static final Serializer INSTANCE = new OpenMetrics100Serializer();
+
+    @Override
+    String headerName(String name, PrometheusType type) {
+      return name;
+    }
+
+    @Override
+    void writeHelp(Writer writer, String description) throws IOException {
+      writeEscapedLabelValue(writer, description);
+    }
+
+    @Override
+    void writeTimestamp(Writer writer, long timestampNanos) throws IOException {
+      long timestampMillis = TimeUnit.NANOSECONDS.toMillis(timestampNanos);
+      writer.write(Long.toString(timestampMillis / 1000));
+      writer.write(".");
+      long millis = timestampMillis % 1000;
+      if (millis < 100) {
+        writer.write('0');
+      }
+      if (millis < 10) {
+        writer.write('0');
+      }
+      writer.write(Long.toString(millis));
+    }
+
+    @Override
+    void writeExemplar(
+        Writer writer, Collection<ExemplarData> exemplars, double minExemplar, double maxExemplar)
+        throws IOException {
+      for (ExemplarData exemplar : exemplars) {
```

Comment:
I believe this code should also be writing dropped attributes as exemplar labels (after your trace_id/span_id code, with a limit on total size of labels written to prometheus).

Reply:
Don't think we do that right now

https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/MetricAdapter.java#L327

Would like to keep this pr to just reproducing our current behavior and we can add features in the future